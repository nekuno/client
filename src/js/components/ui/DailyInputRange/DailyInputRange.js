import PropTypes from 'prop-types';
import React, { Component } from 'react';
import translate from '../../../i18n/Translate';
import Chip from '../Chip/Chip.js';
import styles from './DailyInputRange.scss';

@translate('DailyInputRange')
export default class DailyInputRange extends Component {

    static propTypes = {
        id            : PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
        data          : PropTypes.array,
        onClickHandler: PropTypes.func.isRequired
    };

    constructor(props) {
        super(props);

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(option) {
        const {id, data} = this.props;
        const index = data.indexOf(option);
        let newData;

        if (index !== -1 && data.length > 1) {
            newData = data.filter(singleData => singleData !== option);
        } else if (index !== -1) {
            newData = data;
        } else {
            newData = [...data, option];
        }

        this.props.onClickHandler(id, newData);
    }

    render() {
        const {data, strings} = this.props;
        const options = [
            {
                id: 'morning',
                text: strings.morning
            },
            {
                id: 'afternoon',
                text: strings.afternoon
            },
            {
                id: 'night',
                text: strings.night
            }
        ];

        return (
            <div className={styles.dailyInputRange}>
                {options.map(rangeOption =>
                    <div className={styles.rangeOption} key={rangeOption.id}>
                        <Chip text={rangeOption.text} value={rangeOption.id} selected={data.some(singleData => singleData === rangeOption.id)} fullWidth={true} onClickHandler={this.handleClick}/>
                    </div>
                )}
            </div>
        );
    }
}

DailyInputRange.defaultProps = {
    strings: {
        morning         : 'Morning',
        afternoon       : 'Afternoon',
        night           : 'Night',
        continue        : 'Save & continue'
    }
};