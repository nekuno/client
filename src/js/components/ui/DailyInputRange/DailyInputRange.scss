@import '../../../../scss/_partials/colors';

.daily-input-range {
  padding: 0 12px;

  .range-option {
    display: inline-block;
    width: 33%;
  }
}