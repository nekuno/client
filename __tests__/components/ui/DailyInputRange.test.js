import React from 'react';
import DailyInputRange from '../../../src/js/components/ui/DailyInputRange/DailyInputRange.js';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test DailyInputRange component', () => {
    const mockFn = jest.fn();
    const dailyInputRange = shallow(<DailyInputRange data={['morning', 'night']} onClickHandler={mockFn}/>);

    it('should be defined', () => {
        expect(DailyInputRange).toBeDefined();
    });
    it('should render correctly', () => {
        expect(dailyInputRange).toMatchSnapshot();
    });
});
