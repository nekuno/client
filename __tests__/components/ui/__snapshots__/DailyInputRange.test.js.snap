// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test DailyInputRange component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TranslationComponent
    data={
      Array [
        "morning",
        "night",
      ]
    }
    onClickHandler={[MockFunction]}
    strings={
      Object {
        "afternoon": "Afternoon",
        "continue": "Save & continue",
        "morning": "Morning",
        "night": "Night",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "data": Array [
        "morning",
        "night",
      ],
      "locale": undefined,
      "onClickHandler": [MockFunction],
      "strings": Object {
        "afternoon": "Afternoon",
        "continue": "Save & continue",
        "morning": "Morning",
        "night": "Night",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "data": Array [
          "morning",
          "night",
        ],
        "locale": undefined,
        "onClickHandler": [MockFunction],
        "strings": Object {
          "afternoon": "Afternoon",
          "continue": "Save & continue",
          "morning": "Morning",
          "night": "Night",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
